src/VarDumper.php

```diff
@@ -59,6 +58,125 @@ public static function dumpAsString($var, int $depth = 10, bool $highlight = fal
         return self::$output;
     }
 
+    public static function dumpAsJson($var, int $depth = 50): string
+    {
+        self::$depth = $depth;
+        self::buildVarObjectsCache($var);
+
+        return json_encode(self::dumpNestedInternal($var, 0));
+    }
+
+    public static function dumpObjectsAsJson(): string
+    {
+        $objects = self::$objects;
+        $json = self::dumpNestedInternal($objects, 0, 1);
+
+        return json_encode(self::getObjectsMap($json));
+    }
+
+    /**
+     * Exports a variable as a string representation.
+     *
+     * The string is a valid PHP expression that can be evaluated by PHP parser
+     * and the evaluation result will give back the variable value.
+     *
+     * This method is similar to `var_export()`. The main difference is that
+     * it generates more compact string representation using short array syntax.
+     *
+     * It also handles objects by using the PHP functions serialize() and unserialize().
+     *
+     * PHP 5.4 or above is required to parse the exported value.
+     *
+     * @param mixed $var the variable to be exported.
+     * @return string a string representation of the variable
+     */
+    public static function export($var): string
+    {
+        self::$output = '';
+        self::exportInternal($var, 0);
+        return self::$output;
+    }
+
+    private static function buildVarObjectsCache($var, int $level = 0): void
+    {
+        if (is_array($var)) {
+            if (self::$depth <= $level) {
+                return;
+            } else {
+                foreach ($var as $key => $value) {
+                    self::buildVarObjectsCache($value, $level + 1);
+                }
+            }
+        } elseif (is_object($var)) {
+            if ((($id = array_search($var, self::$objects, true)) !== false) || (self::$depth <= $level)) {
+                return;
+            } else {
+                array_push(self::$objects, $var);
+                if ('__PHP_Incomplete_Class' !== get_class($var) && method_exists($var, '__debugInfo')) {
+                    $dumpValues = $var->__debugInfo();
+                    if (!is_array($dumpValues)) {
+                        throw new \Exception('__debugInfo() must return an array');
+                    }
+                } else {
+                    $dumpValues = (array)$var;
+                }
+                foreach ($dumpValues as $key => $value) {
+                    self::buildVarObjectsCache($value, $level + 1);
+                }
+            }
+        }
+
+        return;
+    }
+
+    private static function dumpNestedInternal($var, int $level, int $objectCollapseLevel = 0)
+    {
+        $output = [];
```

Comment:
Will be nice to move the last else statement to here ;)

Reply:
In what way?

Reply:
Because it looks useless: firstly set variable value to empty array and next set the value from paramaters